langmeta/langmeta/shared/src/main/scala/org/langmeta/semanticdb/Denotation.scala

```diff
@@ -108,6 +134,18 @@ object Denotation extends AbstractFunction4[Long, String, String, List[ResolvedN
             tpe: Option[s.Type]): Denotation =
     new Denotation(flags, name, signature, names, members, overrides, tpe)
 
+  def apply(flags: Long,
```

Comment:
Why do we still bother with apply when we can use the primary ctor?

Reply:
I don't know. For me it's inertia. I'm fine with either.

Reply:
I think we we should not bother with apply anymore.

Reply:
as in, we should not keep on adding new apply constructors. What we should have done in v3.0.0 was to remove all the boilerplate and add `.withAnnotations()` builder methods.